db/event_handler.go

```diff
@@ -68,25 +74,18 @@ func NewWebhook(url string, filterFnString string, timeout *uint64) (*Webhook, e
 // on the event type.
 func (wh *Webhook) HandleEvent(event Event) bool {
 
+	const contentType = "application/json"
 	var payload []byte
-	var contentType string
-	if wh.filter != nil {
-		// If filter function is defined, use it to determine whether to post
-		success, err := wh.filter.CallValidateFunction(event)
-		if err != nil {
-			base.Warnf("Error calling webhook filter function: %v", err)
-		}
-
-		// If filter returns false, cancel webhook post
-		if !success {
-			return false
-		}
-	}
 
 	// Different events post different content by default
 	switch event := event.(type) {
 	case *DocumentChangeEvent:
-		contentType = "application/json"
+		// skip event if this is for a non-winning rev and the winning rev only option is enabled
+		if !event.WinningRevChange && wh.options != nil {
+			if val, ok := wh.options[EventOptionDocumentChangedWinningRevOnly]; ok && val.(bool) {
```

Comment:
Can we do the existence check/map lookup for wh.options[EventOptionDocumentChangedWinningRevOnly] once, in NewWebhook, and store as boolean, instead of checking on every HandleEvent?